docs/remote-cli.md

```diff
@@ -0,0 +1,92 @@
+# Controlling `headscale` with remote CLI
+
+## Prerequisit
+
+- A workstation to run `headscale` (could be Linux, macOS, other supported platforms)
+- A `headscale` server (version `0.13.0` or newer)
+- Access to create API keys (local access to the `headscale` server)
+- `headscale` _must_ be served over TLS/HTTPS with a _trusted_ certificate
```

Comment:
> Why couldn't we use SSL termination on another host like a reverse proxy ?

hmm, maybe this isnt clear, it does use the same certificate that `headscale` already uses, e.g. Let's Encrypt. 

I suppose we could make it allowed to set gRPC in "insecure" mode so it can be proxied through Nginx. My main concern with this is support, I dont run Nginx, and officially supporting it seems out of scope.

Happy for someone to add a config option for that and test it.

> Also headscale is already providing a public key that can be used to send information to it, under /key I believe. It would be easier to setup, assuming that the communications to headscale are already encrypted.

You are right, but [gRPC does not seem to support "bring your own" transport security](https://grpc.io/docs/guides/auth/#supported-auth-mechanisms) (and I dont think I would want to make one and be responsible for it).

I might be missing something, I am reasonably new to gRPC and that world.

Reply:
Actually, have a look at the last couple of commits, I have made insecure available in two ways to accommodate different setups:

1. Allow insecure on the CLI side to _not verify the cert ca_
2. Allow h2c (HTTP2 cleartext) gRPC on the server, to allow for a reverse proxy to terminate the TLS

I have also added an example on how this can be done with Nginx, and actually combining so both gRPC and HTTPS run on the same port (e.g. 443).

Reply:
Thank you very much ! I think this should ease setup ! For custom transport layer, this was just a thought that could be convenient, no need to go there if it's not supported by default !